src/user.ts

```diff
@@ -0,0 +1,23 @@
+// const <name>: <Typ> = <Inhalt>
+const possibleTitles: ['Dr.', 'Prof.', null]=['Dr.', 'Prof.', null];
+export type Title = typeof possibleTitles;
+
+//An interface for user-data
+export interface User {
+  title: Title;
+  firstName: string;
+  lastName: string;
+  readonly joinedAt?: number; //must be Unix Timestamp (generated with Date.now)
+}
```

Comment:
It is good to create an interface to define interfaces between classes and services (and so on). Especially in TypeScript, interfaces are "syntactic sugar": They are going to be removed after compiled to JS. Therefore, you have two options: 
1. You can convert this interface to a class and let it "self-assign" the data
2. You can implement a "build"-function, which creates an instance of the `User`-interface